File: frontend/earthquake_frontend/src/api/types/earthquake.ts
import type { MagnitudeCategory } from '../../util/magnitude.ts';

export interface Earthquake {
    id: number;
    usgsId: string;
    magnitude: number | null;
    magType: string | null;
    place: string | null;
    title: string | null;
    time: string;
    latitude: number | null;
    longitude: number | null;
    depth: number | null;
}

export interface PageResponse<T> {
    content: T[];
    page: number;
    size: number;
    totalElements: number;
    totalPages: number;
}

export interface EarthquakeFilters {
    minMagnitude?: number;
    categories?: MagnitudeCategory[];
    from?: string;
    to?: string;
}

File: backend/earthquake_backend/src/main/java/mk/earthquake_backend/model/domain/Earthquake.java
package mk.earthquake_backend.model.domain;

import jakarta.persistence.*;
import lombok.*;

import java.time.Instant;

@Entity
@Getter
@Builder
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Table(name = "earthquakes")
public class Earthquake {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "usgs_id", nullable = false, unique = true)
    private String usgsId;

    @Column(name = "magnitude")
    private Double magnitude;

    @Column(name = "mag_type", length = 20)
    private String magType;

    @Column(name = "place")
    private String place;

    @Column(name = "title")
    private String title;

    @Column(name = "time", nullable = false)
    private Instant time;

    @Column(name = "latitude")
    private Double latitude;

    @Column(name = "longitude")
    private Double longitude;

    @Column(name = "depth")
    private Double depth;
}
